### PR TITLE
Test

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: .cirun.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+runners:
+  - name: long-runner
+    cloud: aws
+    instance_type: c5d.4xlarge
+    machine_image: ami-0048c368c80e3ee99
+    preemptible: false
+    workflow: .github/workflows/conda-build.yml
+    count: 2
```
Comment (isuruf) [inline .cirun.yml:8]: @aktech, I tried a few different instance_types and they all fail with `no space left on device` while pulling the docker image. Any ideas?

Comment (aktech) [reply .cirun.yml:8]: Yes, that's expected with docker images. The default size of instance is less than 10 GB, which is not really enough.  

I can potentially add the setting/configuration in the `.cirun.yml` to increase disk size but that depends on disk type etc and there are lot of permutations for the same.

So, I would suggest to create a custom AMI (say on top of Ubuntu 20.04) with disk size something like say 30 GB or more and use that AMI ID.

**Comment by wolfv** [comment]
Only using 13% of CPU 😢 

Screenshot 2022-02-08 at 15 49 27

**Comment by isuruf** [comment]
That's probably because of https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/master/recipe/run_conda_forge_build_setup_linux#L27

**Comment by wolfv** [comment]
I thought it's because of `pip` as a build system not supporting multi-core compilation? Doesn't the line you linked just set a default?

**Comment by isuruf** [comment]
> I thought it's because of pip as a build system not supporting multi-core compilation? 

pytorch uses pip, but it calls cmake under the hood.

> Doesn't the line you linked just set a default?

Yes, the default of 2 is not good

**Comment by isuruf** [comment]
It is if the user doesn't set it. We set it there, because usually the value set up by conda-build is wrong. What happens is that the VM that CI gives us will run out of memory or some other resource. We can make an exception for github_actions just like we do for drone.

**Comment by wolfv** [comment]
I don't remember having these issues when running `build_locally.py` on large machines (e.g. on the OVH machine that I used for TensorFlow).

**Comment by isuruf** [comment]
> Should we stop this for now?

sure

> I don't remember having these issues when running build_locally.py on large machines (e.g. on the OVH machine that I used for TensorFlow).

No clue. It might be that tensorflow recipe is not configured to use `CPU_COUNT` variable.

**Comment by isuruf** [comment]
@wolfv, shall we start another build to see if the CPU_COUNT fix is working now?

**Comment by wolfv** [comment]
yes!

**Comment by wolfv** [comment]
maybe we can remove some configs here? We have a 256 core limit

**Comment by isuruf** [comment]
> maybe we can remove some configs here? We have a 256 core limit

I'm not sure I understand. core limit in what?

**Comment by wolfv** [comment]
We (I) have a limit of 256 vCPUs on AWS. 

In general, I just think it might be enough to run 2 or so test configs in parallel.

**Comment by wolfv** [comment]
Anyways, let's just let it run. I am excited :)

**Comment by isuruf** [comment]
@aktech, is there a way to limit the number of runners run in parallel?

**Comment by wolfv** [comment]
I was thinking about removing some configurations from this test build. But it doesn't matter that much for now.

**Comment by aktech** [comment]
> @aktech, is there a way to limit the number of runners run in parallel?

Not from the Cirun interface at the moment, but since I am working on retry mechanism, the failed runner creations would retry and will create runners eventually, when the running jobs get finished. If that helps?

**Comment by isuruf** [comment]
@wolfv, started a build.

@aktech, yeah, that might work. Thanks for the info. I just realized there's a `count` parameter, but I'm not sure what it does and doesn't seem to be documented.

**Comment by aktech** [comment]
> Thanks for the info. I just realized there's a count parameter, but I'm not sure what it does and doesn't seem to be documented.

It's an old function: https://github.com/AktechLabs/cirun-docs/pull/8 It has been removed now as it's not required anymore. (apologies for the fact that it didn't followed any deprecation cycle or documentation update)

So, this is due to the previous webhook, I have been using called as `check_run` which is a single webhook event for the whole workflow, that's why there was a need to creating multiple VMs based on the count, but since now I am using the new `workflow_job` webhook, the `count` parameter is not required any more, as each job has a separate webhook event.

**Comment by wolfv** [comment]
Hmm, AWS still shows only ~15% utilization. I don't know enough about AWS to judge wether this is normal / expected or not and I can't log in to the machines to check with `htop`.

**Comment by aktech** [comment]
Hey @wolfv you can easily log onto machine to check with htop via: https://github.com/marketplace/actions/debugging-with-tmate

**Comment by wolfv** [comment]
still seems really slow. on the other hadn IDK what the expected build time is ...

**Comment by wolfv** [comment]
@aktech do you know if we should expect close to 100% CPU usage on AWS wiht a multicore build?

**Comment by wolfv** [comment]
@isuruf these builds are still using MAX_JOBS=2 ( https://github.com/conda-forge/pytorch-cpu-feedstock/runs/5181076142?check_suite_focus=true#step:3:2075 )

**Comment by isuruf** [comment]
> To create a local Git repository instead, add Git 2.18 or higher to the PATH

Looks like we need an image with Git 2.18 or higher for actions/checkout to get a git repo.

**Comment by aktech** [comment]
@isuruf @wolfv Do you need any help with image creation here or with CI in general?

**Comment by isuruf** [comment]
@aktech, yes. Help with image creation is greatly appreciated.

**Comment by aktech** [comment]
I'll be happy to create the image, should I create it on my AWS account or can I have access to the AWS account which you guys are using?

**Comment by isuruf** [comment]
> I'll be happy to create the image, should I create it on my AWS account or can I have access to the AWS account which you guys are using?

Thanks for the offer. I think it's best to create it on your account as I still need to figure out if I can give permissions to you.

**Comment by aktech** [comment]
Hey @isuruf I created the image with latest Git (2.35.1) on top of the Ubuntu 18.04 with Docker (ami-0048c368c80e3ee99).

Here are the job runs on my fork: https://github.com/aktech/pytorch-cpu-feedstock/actions/runs/1952646838

The AMI ID is `ami-0febc229056b2de48` in `eu-west-1` (Ireland region). I have made it public, I believe you can use it as well.

**Comment by aktech** [comment]
Both jobs [completed successfully.](https://github.com/aktech/pytorch-cpu-feedstock/actions/runs/1952646838)

File: .cirun.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+runners:
+  - name: long-runner
+    cloud: aws
+    instance_type: c5d.4xlarge
+    machine_image: ami-0582cf60af51d3113
```
Comment (aktech) [inline .cirun.yml:5]:
```suggestion
    machine_image: ami-0febc229056b2de48
```

File: .cirun.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    preemptible: false
+    workflow: .github/workflows/conda-build.yml
+    count: 2
```
Comment (aktech) [inline .cirun.yml:8]:
```suggestion
```

**Comment by Tobias-Fischer** [comment]
Can we close here @isuruf?

**Pull request closed without merging**